### ISSUE TITLE
Random NoneType error while downloading

### ISSUE THREAD

**Issue by shinji257** [mansuf/mangadex-downloader#18]
Version 1.0.0 (I just installed it today)
Install via Python PIP
OS: Linux Mint 20.3
Python Version: 3.8.10
Command: mangadex-dl --login-username (redacted) --login-password (redacted) --save-as tachiyomi --cover original --unsafe https://mangadex.org/title/c1d990dd-9ff2-4e47-b93d-5e56e34c5069/digimon-xros-wars

I know tachiyomi is supposed to be default but I was seeing it using raw format so I enforced it.

I'm not sure what is triggering this.  If I restart the download it will go down and skip the files it grabbed then pickup just fine where it left off.

```
[ERROR] Unhandled exception, AttributeError: 'NoneType' object has no attribute 'startswith'
Traceback (most recent call last):
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/__init__.py", line 35, in _main
    download(args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/download.py", line 12, in download
    url(args, args.type)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/url.py", line 144, in __call__
    self.func(self.id, args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/url.py", line 66, in download_manga
    dl_manga(*args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/main.py", line 331, in download
    fmt.main()
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/format/tachiyomi.py", line 61, in main
    success = downloader.download()
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/downloader.py", line 172, in download
    self._report(resp, length, round((t2 - t1) * 1000), False)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/downloader.py", line 254, in _report
    if resp.headers.get('x-cache').startswith('HIT'):
AttributeError: 'NoneType' object has no attribute 'startswith'

```

**Comment by mansuf** [comment]
> I'm not sure what is triggering this. If I restart the download it will go down and skip the files it grabbed then pickup just fine where it left off.

Server error, which caused downloader report system is failing. i have should expect that Response headers are not parsed properly when this happen.

Whenever you logged in and downloading a manga, mangadex-downloader automatically reports to MangaDex network for each images that have you downloaded ([read more here](https://api.mangadex.org/docs/reading-chapter/#report)). I suggest you to not login until this issue is fixed

**Comment by shinji257** [comment]
Ok.  I stopped logging in and I still got the error randomly.  Usually I can go ahead and restart the download and be fine.

**Comment by mansuf** [comment]
Can you provide full logs ?

**Comment by shinji257** [comment]
I don't have any logs to give at the moment (it just shows the files it is downloading then suddenly error) but I'll enable verbose mode and have it start piping to a file for the next time it hits.

**Comment by mansuf** [comment]
I have created pull request regarding this issue, it should be fixed. You can test it by install from repository [bug/18](https://github.com/mansuf/mangadex-downloader/tree/bug/18). Let me know if this issue still happens.

**Comment by shinji257** [comment]
> I have created pull request regarding this issue, it should be fixed. You can test it by install from repository [bug/18](https://github.com/mansuf/mangadex-downloader/tree/bug/18). Let me know if this issue still happens. I will merge the pull request to main repository once this issue is fixed

Ok.  Previous reply was before I saw your PR so I'm going to test it some more and see if I hit it with the fix.

**Comment by shinji257** [comment]
Really confident that the issue is cleared with the PR.  I have been downloading near non-stop and have been unable to reproduce it.  Now that I'm posting this I'm sure I'll come back in a few minutes and report that it is still there but if I don't...

**Comment by mansuf** [comment]
Well, it's now more than a hours since your last reply. I will merge the PR and mark this issue as fixed.

**Comment by mansuf** [comment]
This fix is released in [v1.0.1](https://github.com/mansuf/mangadex-downloader/releases/tag/v1.0.1)
